roofit/roofit/src/RooLagrangianMorphFunc.cxx

```diff
@@ -0,0 +1,3558 @@
+  //          idx << ")" << std::endl;
+  //        } else {
+  //          //coutD(ObjectHandling) << obj2->GetName() << " is non-zero (idx="
+  //          << idx << ")" << std::endl;
+  //        }
+  //        idx++;
+  //      }
+  //    }
+
+  FormulaList formulas;
+  for (size_t i = 0; i < morphfunc.size(); ++i) {
+    RooArgList ss;
+    bool isZero = false;
+    std::string reason;
+    // check if this is a blacklisted interference term
+    for (const auto &group : nonInterfering) {
+      int nInterferingOperators = 0;
+      for (size_t j = 0; j < morphfunc[i].size(); ++j) {
+        if (morphfunc[i][j] % 2 == 0)
+          continue; // even exponents are not interference terms
+        if (group->find(couplings.at(j)
+                            ->GetName())) { // if the coupling is part of a
+                                            // "pairwise non-interfering group"
+          nInterferingOperators++;
+        }
+      }
+      if (nInterferingOperators > 1) {
+        isZero = true;
+        reason = "blacklisted interference term!";
+      }
+    }
+    int nNP = 0;
+    if (!isZero) {
+      // prepare the term
+      for (size_t j = 0; j < morphfunc[i].size(); ++j) {
+        const int exponent = morphfunc[i][j];
+        if (exponent == 0)
+          continue;
+        RooAbsReal *coupling = dynamic_cast<RooAbsReal *>(couplings.at(j));
+        for (int k = 0; k < exponent; ++k) {
+          ss.add(*coupling);
+          if (coupling->getAttribute("NP")) {
+            nNP++;
+          }
+        }
+        std::string cname(coupling->GetName());
+        if (coupling->getAttribute("LO") && exponent > 1) {
+          isZero = true;
+          reason = "coupling " + cname + " was listed as leading-order-only";
+        }
+        // mark the term as zero if any of the couplings are zero
+        if (!isZero && couplingsZero[j]) {
+          isZero = true;
+          reason = "coupling " + cname + " is zero!";
+        }
+      }
+    }
+    // check and apply flags
+    bool removedByFlag = false;
+    RooAbsReal *obj;
+
+    for (auto itr : flags) {
+      obj = dynamic_cast<RooAbsReal *>(itr);
+      if (!obj)
+        continue;
+      TString sval(obj->getStringAttribute("NP"));
+      int val = atoi(sval);
+      if (val == nNP) {
+        if (flagsZero.find(obj->GetName()) != flagsZero.end() &&
+            flagsZero.at(obj->GetName())) {
+          removedByFlag = true;
+          reason = Form("flag %s is zero", obj->GetName());
+        }
+        ss.add(*obj);
+      }
+    }
+
+    // create and add the formula
+    if (!isZero && !removedByFlag) {
+      // build the name
+      const TString name = Form("%s_pol%lu", mfname, i);
+      //        formulas[i] = new RooProduct(name.Data(),::concatNames(ss," *
+      //        ").c_str(),ss);
+      formulas[i] = std::make_unique<RooProduct>(
+          name.Data(), ::concatNames(ss, " * ").c_str(), ss);
```

Comment:
```suggestion
      const auto name = std::string(mfname) + "_pol" + std::to_string(i);
      //        formulas[i] = new RooProduct(name.Data(),::concatNames(ss," *
      //        ").c_str(),ss);
      formulas[i] = std::make_unique<RooProduct>(
          name.c_str(), ::concatNames(ss, " * ").c_str(), ss);
```
Please use `std::to_string` here, then we avoid the warning in the Debian build.